service/matching/taskListManager.go

```diff
@@ -618,6 +624,21 @@ func (c *taskListManagerImpl) getIsolationGroupForTask(ctx context.Context, task
 	return defaultTaskBufferIsolationGroup, nil
 }
 
+func (c *taskListManagerImpl) getPollerIsolationGroups() []string {
+	groupSet := c.pollerHistory.getPollerIsolationGroups(time.Now().Add(-10 * time.Second))
```

Comment:
Mind adding a comment about what 10 seconds comes from?

Reply:
Ah, I hadn't read the code in func `(c *taskListManagerImpl) HasPollerAfter` closely enough, thank you for the correction. I had the wrong picture in my head regarding the risk of missing the poller. this historical lookback is used in addition to a map of outstanding polls.